packages/netsuite-adapter/src/filters/workflow_account_specific_values.ts

```diff
@@ -0,0 +1,768 @@
+
+  return _(suiteQLTableInstances)
+    .keyBy(instance => instance.elemID.name)
+    .mapValues(instance => instance.value[INTERNAL_IDS_MAP] as InternalIdsMap)
+    .mapValues(
+      internalIdsMap => _(internalIdsMap)
+        .entries()
+        .groupBy(row => row[1].name)
+        .mapValues(rows => rows.map(row => row[0]))
+        .value()
+    )
+    .value()
+}
+
+const toMissingInternalIdWarning = (
+  elemID: ElemID,
+  name: string,
+  field: string
+): ChangeError => ({
+  elemID,
+  severity: 'Warning',
+  message: 'Could not identify value in workflow',
+  detailedMessage: `Could not find object "${name}" for field "${field}". Setting ACCOUNT_SPECIFIC_VALUE instead.`,
+})
+
+const toMultipleInternalIdsWarning = (
+  elemID: ElemID,
+  name: string,
+  internalId: string
+): ChangeError => ({
+  elemID,
+  severity: 'Warning',
+  message: 'Multiple objects with the same name',
+  detailedMessage: `There are multiple objects with the name "${name}". Using the first one (internal id: ${internalId}).`,
+})
+
+const resolveAccountSpecificValues = (
+  instance: InstanceElement,
+  suiteQLTablesMap: Record<string, Record<string, string[]>>,
+): ChangeError[] => {
+  const resolveWarnings: ChangeError[] = []
+  const resolveAccountSpecificValueForField = (
+    path: ElemID,
+    value: Values,
+    field: string,
+    fieldValue: unknown,
+  ): void => {
+    if (typeof fieldValue !== 'string' || !isResolvedAccountSpecificValue(fieldValue)) {
+      return
+    }
+    const name = getResolvedAccountSpecificValue(fieldValue)
+    const fieldType = getQueryRecordFieldType(instance, value, field, suiteQLTablesMap, {})
+    const internalIds = _.uniq(makeArray(fieldType).flatMap(type => suiteQLTablesMap[type][name] ?? []))
+    const internalId = internalIds[0]
+    if (internalIds.length === 0) {
+      log.warn(
+        'could not find internal id of "%s" in field %s from value: %o',
+        name, path.createNestedID(field).getFullName(), value,
+      )
+      resolveWarnings.push(toMissingInternalIdWarning(path, name, field))
+      value[field] = ACCOUNT_SPECIFIC_VALUE
+    } else {
+      if (internalIds.length > 1) {
+        log.warn('there are several internal ids for name %s - using the first: %d', name, internalId)
+        resolveWarnings.push(toMultipleInternalIdsWarning(path, name, internalId))
+      }
+      value[field] = internalId
+    }
+  }
+
+  walkOnElement({
+    element: instance,
+    func: ({ value, path }) => {
+      if (!_.isPlainObject(value)) {
+        return WALK_NEXT_STEP.RECURSE
+      }
+      Object.entries(value).forEach(([field, fieldValue]) => {
+        resolveAccountSpecificValueForField(path, value, field, fieldValue)
+      })
+      return WALK_NEXT_STEP.RECURSE
+    },
+  })
+
+  return resolveWarnings
+}
+
+export const handleWorkflowAccountSpecificValuesOnDeploy = async (
+  instances: InstanceElement[],
+  elementsSource: ReadOnlyElementsSource,
+): Promise<ChangeError[]> => {
```

Comment:
I think that the naming here is not indicative enough.
Maybe resolveWorkflowsAccountSpecificValues would be better? WDYT?
```suggestion
export const resolveWorkflowsAccountSpecificValues = async (
  workflowInstances: InstanceElement[],
  elementsSource: ReadOnlyElementsSource,
): Promise<ChangeError[]> => {
```